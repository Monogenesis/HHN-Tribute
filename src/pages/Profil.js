import React, { useEffect, useState } from "react";
import ReactDOM from "react-dom";
import "./pagestyles/Profil.css";
import ReturnDataButton from "../components/ReturnData/ReturnData";
import DataDonation from "./DataDonation";
import GatherBaseData from "../components/GatherBaseData/GatherBaseData.js";
import { CheckRegistered } from "../components/RegistrationCheck/RegistrationCheck";
import { getCurrentUser } from "../components/UserAuthentication/UserAuthentication";
import HealthTrackerButton from "../components/HealthTrackerButton/HealthTrackerButton";
import TransmissionIndicator from "../components/TransmissionIndicator/TransmissionIndicator";

import { IconButton, Tooltip } from "@material-ui/core";
import { FcInfo } from "react-icons/fc";
import { CgData } from "react-icons/cg";
import { IconContext } from "react-icons";

const firebase = require("firebase");
require("firebase/firestore");

const db = firebase.firestore().collection("userdata");
let currentDate = new Date().toISOString().slice(0, 10);

var userID;
var currentGender;
var currentAge;
var currentWeight;
var currentHeight;
var currentZipCode;
var lastTimestamp;

const Profil = () => {
  useEffect(() => {
    document.getElementById("newDataDonation").disabled = true;
  }, []);
  return (
    <header>
      <section className="Profil">
        <div id="root">
          <section className="ProfileSection">
            <h1 className="ProfilTitle">Datenspende</h1>
          </section>
          <section className="UserIDSection">
            <h2 className="UserIDTitle">
              <Tooltip
                title={
                  <span>
                    <h3>
                      Hier können Sie ihr persönliches Pseudonym einsehen und
                      ihre Stammdaten aktualisieren. Das Pseudonym dient dazu
                      ihre Identität zu verschleiern.
                    </h3>
                    <br />
                    <h3>
                      Sie können auch eine Pseudoverbindung zu einem
                      Gesundheitstracker herrstellen, um uns regelmäßig
                      zusätztliche Daten wie z.B. Körpertemperatur und
                      Herzfrequenz zu spenden.
                    </h3>
                    <br />
                    <h3>
                      Wenn Sie dieses Symbol
                      <IconContext.Provider
                        value={{ color: "#7CFC00", size: "20px" }}
                      >
                        <CgData></CgData>
                      </IconContext.Provider>
                      in der oberen Leiste sehen, werden Daten auf unsere
                      Datenbank geschrieben.
                    </h3>
                    <br />
                    <h3>
                      Diese Daten werden zufällig generiert und dienen dazu
                      andere Funktionen wie die Heldenkarte mit Daten zu füllen.
                    </h3>
                  </span>
                }
                placement="bottom-start"
                enterDelay={0}
                enterTouchDelay={50}
                leaveTouchDelay={30000}
                arrow="true"
              >
                <IconButton>
                  <FcInfo size={50} />
                </IconButton>
              </Tooltip>
              Pseudonym:
            </h2>
            <p></p>
            <article className="UserID" id="UserID">
              ******************
            </article>
            <p className="Space"></p>
          </section>
          <section className="MetaDatenSection">
            <h3 className="MetadatenTitle">Ihre Stammdaten:</h3>
            <article className="Gender" id="currentGender">
              ------
            </article>
            <p></p>
            <article className="Age" id="currentAge">
              ------
            </article>
            <p></p>
            <article className="Height" id="currentHeight">
              ------
            </article>
            <p></p>
            <article className="Weight" id="currentWeight">
              ------
            </article>
            <p></p>
            <article className="ZipCode" id="currentZipCode">
              ------
            </article>
            <p className="Space2"></p>
          </section>
          <section className="TimeStampSection">
            <h4 className="TimestampTitle">Letzte Aktualisierung:</h4>
            <article className="Date" id="lastTimestamp">
              ------
            </article>
          </section>
          <div className="tableDiv">
            <div className="centerDiv">
              <table className="smartDeviceTable">
                <tr className="buttonRow">
                  <td>
                    <button
                      type="button"
                      id="newDataDonation"
                      href="/datenspenden"
                      onClick={goToRegistration}
                    >
                      Stammdaten aktualisieren
                    </button>
                    <div class="toggle-container">
                      <HealthTrackerButton />
                    </div>
                  </td>
                </tr>
              </table>
            </div>
          </div>
        </div>
      </section>
    </header>
  );
};

/**
 * Retrieves and sets the total user count value that is stored on firebase.
 */
export async function SetMetaData(id) {
  userID = id;
  await db
    .doc(id)
    .collection("basedata")
    .orderBy("timestamp", "desc")
    .limit(1)
    .get()
    .then((col) => {
      col.forEach(function (doc) {
        currentGender = doc.data().gender;
        currentAge = doc.data().age;
        currentWeight =
          doc.data().weight == "Keine Angabe"
            ? "Keine Angabe"
            : doc.data().weight + "kg";
        currentHeight =
          doc.data().height == "Keine Angabe"
            ? "Keine Angabe"
            : doc.data().height + "cm";
        currentZipCode = doc.data().zipcode;
        lastTimestamp = doc.data().timestamp;
      });
    })
    .then(() => {
      UpdateMetaDataLabel();
    });
}

/**
 * Updates the 'MetaDataLabels' with a new value.
 */
const UpdateMetaDataLabel = () => {
  try {
    document.getElementById("newDataDonation").disabled = false;

    document.getElementById("UserID").innerHTML = userID;
    document.getElementById("currentGender").innerHTML =
      "Geschlecht: " + currentGender;
    document.getElementById("currentAge").innerHTML = "Alter: " + currentAge;
    document.getElementById("currentWeight").innerHTML =
      "Gewicht: " + currentWeight;
    document.getElementById("currentHeight").innerHTML =
      "Größe: " + currentHeight;
    document.getElementById("currentZipCode").innerHTML =
      "Postleitzahl: " + currentZipCode;
    document.getElementById("lastTimestamp").innerHTML = lastTimestamp;
  } catch (error) {
    console.log("UpdateMetaDataLabel() - not on Profil.");
  }
};

const goToRegistration = () => {
  console.log("TESTESTEST");
  ReactDOM.render(<DataDonation />, document.getElementById("root"));
  CheckRegistered(getCurrentUser().uid).then(() => {
    GatherBaseData();
  });
};

export default Profil;
