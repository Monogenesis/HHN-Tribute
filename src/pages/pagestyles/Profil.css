:root {
    --backgroundColor: #1a2833;
    --logoColor: #12dbe2;
    --borderColor: #0e8daa;
    --buttonColor: #0e93b1;
    --navbarColor: #152028;
}

*{
    font-family: 'Montserrat', sans-serif;
  }

.Profil {
    color: var(--backgroundColor);
    background: white;
    height: 85vh;
}

.NavbarElements {
    background: var(--navbarColor);
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.5rem;
    border-bottom: 4px solid (var(--borderColor));
}

.ProfilTitle {
    font-family: 'Montserrat', sans-serif;
    border-bottom: 4px solid var(--logoColor);
    text-align: center;
    padding: 1%;
    font-size: 2.5rem;
}

.ProfileSection {
    color: var(--logoColor);
    background: var(--backgroundColor);
}

.UserIDTitle {
    font-size: 2.3em;
    font-family: 'Montserrat', sans-serif;
    text-align: center;
    padding-top: 15px;
    text-decoration: underline;
}

#UserID {
    text-align: center;
    font-size: 2em;
    font-family: 'Montserrat', sans-serif;
    padding-top: 1%;
    padding-bottom: 1%;
}

.Space, .Space2, .TimeStampSection {
    border-bottom: 4px solid var(--logoColor);
}

.MetadatenTitle {
    font-size: 2.5em;
    font-family: 'Montserrat', sans-serif;
    text-align: center;
    padding-top: 10px;
    padding-bottom: 10px;
    text-align: center;
}

.MetaDatenSection {
    background-color: #1a2833;
    color: var(--logoColor);
}

.Gender, .Age, .Weight, .Height, .ZipCode {
    font-size: 2em;
    text-align: center;
    padding-top: 0.5%;
    padding-bottom: 0.5%;
}

.TimestampTitle {
    font-size: 2.3em;
    font-family: 'Montserrat', sans-serif;
    text-align: center;
    padding-top: 1%;
    text-decoration: underline;
}

.Date {
    font-size: 1.4em;
    font-family: 'Montserrat', sans-serif;
    text-align: center;
    padding-top: 1%;
    padding-bottom: 1%;
    height: 5vh;
}

.smartDeviceTable {
    position: relative;
    text-align: center;
    background-color: var(--backgroundColor);
}

.tableDiv {
    height: 200px;
    position: relative;
    background-color: var(--backgroundColor);
}

.centerDiv {
    margin: 0;
    position: absolute;
    height: 0%;
    top: 0%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
}

#newDataDonation{
    background-color: var(--buttonColor);
    color: white;
    font-family: 'Montserrat', sans-serif;
    padding: 5%;
    font-size: 14px;
    margin: 20px auto 0;
}

.toggle-container {
    padding-top: 5px;
    float: 1px;
    flex-direction: row;
    flex-flow: row;
    font-family: 'Montserrat';
    font-weight: bold;
    color: var(--logoColor);
}

@media screen and (max-width: 960px) {
    
    .ProfilTitle {
        font-size: 1.8em;
        color: var(--logoColor);
        border-bottom: 4px solid var(--logoColor);
    }
    .UserIDTitle {
        font-size: 1.7em;
        padding-top: 15px;
        text-decoration: underline;
    }
    #UserID {
        text-align: center;
        font-size: 1.2em;
        padding-top: 20px;
        padding-bottom: 20px;
    }
    .MetadatenTitle {
        font-size: 1.70em;
        padding-top: 10px;
        padding-bottom: 10px;
        text-decoration: underline;
        text-align: center;
    }
    .Gender, .Age, .Weight, .Height, .ZipCode {
        font-size: 1em;
        padding-top: 10px;
        padding-bottom: 10px;
        letter-spacing: 0px;
    }
    .TimestampTitle {
        font-size: 1.7em;
        padding-top: 15px;
        text-decoration: underline;
    }
    .Date {
        font-size: 20px;
        padding-top: 15px;
        padding-bottom: 10px;
    }
    #newDataDonation{
        background-color: var(--buttonColor);
        color: white;
        font-family: 'Montserrat', sans-serif;
        padding: 5%;
        font-size: 14px;
        margin: 50px auto 0;
    }
}

@media screen and (max-width: 450px) {
    .ProfilTitle {
        font-size: 1.5em;
        border-bottom: 4px solid var(--logoColor);
    }
    .UserIDTitle {
        font-size: 1.2em;
        padding-top: 15px;
        text-decoration: underline;
    }
    #UserID {
        text-align: center;
        font-size: 0.9em;
        padding-top: 20px;
        padding-bottom: 20px;
    }
    .MetadatenTitle {
        font-size: 1.2em;
        padding-top: 10px;
        padding-bottom: 10px;
        text-decoration: underline;
        text-align: center;
    }
    .Gender, .Age, .Weight, .Height, .ZipCode {
        font-size: 0.8em;
        padding-top: 10px;
        padding-bottom: 10px;
        letter-spacing: 0px;
    }
    .TimestampTitle {
        font-size: 1.2em;
        padding-top: 15px;
        text-decoration: underline;
    }
    .Date {
        font-size: 15px;
        padding-top: 15px;
        padding-bottom: 10px;
    }
    .buttonSection {
        size: auto;
    }
    #newDataDonation{
        background-color: var(--buttonColor);
        color: white;
        font-family: 'Montserrat', sans-serif;
        padding: 5%;
        font-size: 14px;
        margin: 50px auto 0;
        margin-left: 6%;
    }
}