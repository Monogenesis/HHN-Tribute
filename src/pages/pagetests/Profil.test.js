import React from "react";
import renderer from "react-test-renderer";
import Profil from "../Profil.js"

test("Profile Page renders correctly", () => {
    const tree = renderer.create(<Profil></Profil>).toJSON();
expect(tree).toMatchSnapshot();
})