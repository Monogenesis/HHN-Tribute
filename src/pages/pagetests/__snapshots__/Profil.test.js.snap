// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page renders correctly 1`] = `
<header>
  <section
    className="Profil"
  >
    <div
      id="root"
    >
      <section
        className="ProfileSection"
      >
        <h1
          className="ProfilTitle"
        >
          Profile
        </h1>
      </section>
      <section
        className="UserIDSection"
      >
        <h2
          className="UserIDTitle"
        >
          Pseudonym:
        </h2>
        <p />
        <article
          className="UserID"
          id="UserID"
        >
          ******************
        </article>
        <p
          className="Space"
        />
      </section>
      <section
        className="MetaDatenSection"
      >
        <h3
          className="MetadatenTitle"
        >
          Your information:
        </h3>
        <article
          className="Gender"
          id="currentGender"
        >
          ------
        </article>
        <p />
        <article
          className="Age"
          id="currentAge"
        >
          ------
        </article>
        <p />
        <article
          className="Weight"
          id="currentWeight"
        >
          ------
        </article>
        <p />
        <article
          className="Height"
          id="currentHeight"
        >
          ------
        </article>
        <p />
        <article
          className="ZipCode"
          id="currentZipCode"
        >
          ------
        </article>
        <p
          className="Space2"
        />
      </section>
      <section
        className="TimeStampSection"
      >
        <h4
          className="TimestampTitle"
        >
          Last Update:
        </h4>
        <article
          className="Date"
          id="lastTimestamp"
        >
          ------
        </article>
      </section>
      <div
        className="tableDiv"
      >
        <div
          className="centerDiv"
        >
          <table
            className="smartDeviceTable"
          >
            <tr
              className="buttonRow"
            >
              <td
                className="returnDataButton"
              >
                <div>
                  <button
                    id="returnDataButton"
                    onClick={[Function]}
                    type="button"
                  >
                    Check my stats
                  </button>
                </div>
              </td>
              <td
                className="connectSmartDeviceButton"
              >
                 
                <div>
                  <button
                    id="smartDeviceButton"
                    onClick={[Function]}
                    type="button"
                  >
                    Connect Device
                  </button>
                </div>
              </td>
            </tr>
          </table>
        </div>
      </div>
    </div>
  </section>
</header>
`;
